source/taup/taup_time.rst

```diff
@@ -31,7 +31,8 @@ TauP 默认台站位于地表，可使用 ``--stadepth`` 选项设置台站深
 
    $ taup_time -h 300 -evt 40 10 -sta -20 -10 -ph P,S,PcP,ScS,PKiKP
 
-在震中距比较小的情况下，可以使用 ``-km`` 选项指定震中距（单位为 km）。
+在震中距比较小的情况下，可以使用 ``-km`` 选项指定震中距（单位为 km）,
+此时 taup_time 假定地球半径为 6371 km。
```

Comment:
Are you sure? I think the radius can be other values, based on the current model used.

Reply:
I am not very sure, but the author use **assume** instead of **default** or other words.

![taup_time](https://user-images.githubusercontent.com/50591376/103458826-b8e50b80-4d46-11eb-98a7-f40bcc680c80.png)

Reply:
https://github.com/crotwell/TauP/blob/ecccb870c87b73ca7150414487e9b4d693a63834/src/main/java/edu/sc/seis/TauP/TauP_Time.java#L613

```
                } else if(dashEquals("km", args[i])) {
                    degrees = Double.valueOf(args[i + 1]).doubleValue() / 6371
                            * 180.0 / Math.PI;
                    i++;
                }
```

It seems your description is correct, but it's a TauP bug, right?

Reply:
I remove the last 1000 km of PREM model to create a custom model.

```
 5149.50    10.35568   0.00000  12.16634   57822.0       0.0
inner-core
 5149.50    11.02827   3.50432  12.76360     445.0      85.0
 5171.00    11.03643   3.51002  12.77493     445.0      85.0
 5271.00    11.07249   3.53522  12.82501     443.0      85.0
 5371.00    11.10542   3.55823  12.87073     440.0      85.0
```

Then I use this model to test.

```bash
$ taup_time -mod custom.nd -h 300 -deg 62.58 -ph P

Model: custom
Distance   Depth   Phase   Travel    Ray Param  Takeoff  Incident  Purist    Purist
  (deg)     (km)   Name    Time (s)  p (s/deg)   (deg)    (deg)   Distance   Name 
-----------------------------------------------------------------------------------
   62.58   300.0   P        509.66     5.674     33.96    20.55    62.58   = P    

$ taup_time -mod custom.nd -h 300 -km 6959 -ph P 

Model: custom
Distance   Depth   Phase   Travel    Ray Param  Takeoff  Incident  Purist    Purist
  (deg)     (km)   Name    Time (s)  p (s/deg)   (deg)    (deg)   Distance   Name 
-----------------------------------------------------------------------------------
   62.58   300.0   P        509.68     5.674     33.96    20.55    62.58   = P    
```

If PREM is used directly,

```bash
$ taup_time -mod prem -h 300 -deg 62.58 -ph P     

Model: prem
Distance   Depth   Phase   Travel    Ray Param  Takeoff  Incident  Purist    Purist
  (deg)     (km)   Name    Time (s)  p (s/deg)   (deg)    (deg)   Distance   Name 
-----------------------------------------------------------------------------------
   62.58   300.0   P        591.87     6.556     32.62    20.00    62.58   = P    

$ taup_time -mod prem -h 300 -km 6959 -ph P     

Model: prem
Distance   Depth   Phase   Travel    Ray Param  Takeoff  Incident  Purist    Purist
  (deg)     (km)   Name    Time (s)  p (s/deg)   (deg)    (deg)   Distance   Name 
-----------------------------------------------------------------------------------
   62.58   300.0   P        591.90     6.555     32.62    20.00    62.58   = P    
```

Reply:
If I understand you correctly, your example confirms that there is a TauP bug, right?

Reply:
> If I understand you correctly, your example confirms that there is a TauP bug, right?

Yes. But the same question comes, the travel time changes.

Reply:
> taup_time actually uses the true radius when calculating travel time.

Of course, that's why TauP can be used to calculate traveltimes for the moon and Mars.

Reply:
Cool~ But on the Earth, almost all the model has a radius of 6371 km. Not a big problem~

Reply:
Let's report a TauP bug first.

Reply:
Okay, may you can report it directly.

Reply:
OK. Will do it.